python/sparkdl/graph/input.py

```diff
@@ -0,0 +1,317 @@
+        """
+        assert signature_def_key is not None
+        return _from_checkpoint_impl(checkpoint_dir, signature_def_key, feed_names=None,
+                                     fetch_names=None)
+
+    @classmethod
+    def fromSavedModel(cls, saved_model_dir, tag_set, feed_names, fetch_names):
+        """
+        Construct a TFInputGraph object from a saved model (`tf.SavedModel`) directory.
+        Ignore the the embedded signature_def, if there is any.
+
+        :param saved_model_dir: str, name of the directory containing the TensorFlow graph
+                                training checkpoint.
+        :param tag_set: str, name of the graph stored in this meta_graph of the saved model
+                        that we are interested in using.
+        :param feed_names: list, names of the input tensors.
+        :param fetch_names: list, names of the output tensors.
+        """
+        return _from_saved_model_impl(saved_model_dir, tag_set, signature_def_key=None,
+                                      feed_names=feed_names, fetch_names=fetch_names)
+
+    @classmethod
+    def fromSavedModelWithSignature(cls, saved_model_dir, tag_set, signature_def_key):
+        """
+        Construct a TFInputGraph object from a saved model (`tf.SavedModel`) directory,
+        using the embedded signature_def. Throw error if we cannot find an entry with
+        the `signature_def_key` inside the `signature_def`.
+
+        :param saved_model_dir: str, name of the directory containing the TensorFlow graph
+                                training checkpoint.
+        :param tag_set: str, name of the graph stored in this meta_graph of the saved model
+                        that we are interested in using.
+        :param signature_def_key: str, name of the mapping contained inside the `signature_def`
+                                  from which we retrieve the signature key to tensor names mapping.
+        """
+        assert signature_def_key is not None
+        return _from_saved_model_impl(saved_model_dir, tag_set, signature_def_key=signature_def_key,
+                                      feed_names=None, fetch_names=None)
+
+
+def _from_checkpoint_impl(checkpoint_dir, signature_def_key, feed_names, fetch_names):
+    """
+    Construct a TFInputGraph from a model checkpoint.
+    Notice that one should either provide the `signature_def_key` or provide both
+    `feed_names` and `fetch_names`. Please set the unprovided values to None.
+
+    :param signature_def_key: str, name of the mapping contained inside the `signature_def`
+                              from which we retrieve the signature key to tensor names mapping.
+    :param feed_names: list, names of the input tensors.
+    :param fetch_names: list, names of the output tensors.
+    """
+    assert (feed_names is None) == (fetch_names is None), \
+        'feed_names and fetch_names, if provided must appear together'
```

Comment:
"must appear together" -> "must be both non-None"

Reply:
@phi-dbq did you miss this comment?